From 42a4a8ae65da1e71503e2aba2a1af82a09283b63 Mon Sep 17 00:00:00 2001
From: Lukas Bezdicka <lbezdick@redhat.com>
Date: Wed, 10 Sep 2014 19:29:29 +0200
Subject: [PATCH] Add manage_service feature

puppet-keystone lacks of disabling service managing. This patch adds
$manage_service parameter to relevant class.

Change-Id: I80a55857442c6cd32387481fbe68b54f52e827a1
Closes-bug: #1359823

Conflicts:
        manifests/init.pp
---
 manifests/init.pp | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index 80d8819..f5b2081 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -38,6 +38,11 @@
 #     Defaults to /var/cache/keystone.
 #   [memcache_servers] List of memcache servers/ports. Optional. Used with
 #     token_driver keystone.token.backends.memcache.Token.  Defaults to false.
+#
+#   [manage_service]
+#      (optional) If Puppet should manage service startup / shutdown.
+#      Defaults to true.
+#
 #   [enabled] If the keystone services should be enabled. Optional. Default to true.
 #
 #   [*database_connection*]
@@ -243,6 +248,7 @@ class keystone(
   $ssl_cert_subject      = '/C=US/ST=Unset/L=Unset/O=Unset/CN=localhost',
   $cache_dir             = '/var/cache/keystone',
   $memcache_servers      = false,
+  $manage_service        = true,
   $enabled               = true,
   $database_connection   = 'sqlite:////var/lib/keystone/keystone.db',
   $database_idle_timeout = '200',
@@ -552,10 +558,12 @@ class keystone(
     }
   }
 
-  if $enabled {
-    $service_ensure = 'running'
-  } else {
-    $service_ensure = 'stopped'
+  if $manage_service {
+    if $enabled {
+      $service_ensure = 'running'
+    } else {
+      $service_ensure = 'stopped'
+    }
   }
 
   if $validate_service {
-- 
1.9.3

